utils/dataloaders.py

```diff
@@ -640,85 +651,128 @@ def cache_labels(self, path=Path('./labels.cache'), prefix=''):
     def __len__(self):
         return len(self.im_files)
 
-    # def __iter__(self):
-    #     self.count = -1
-    #     print('ran dataset iter')
-    #     #self.shuffled_vector = np.random.permutation(self.nF) if self.augment else np.arange(self.nF)
-    #     return self
-
+    def get_image_and_label(self, index):
+        img, (h0, w0), (h, w) = self.load_image(index)
+        bboxes = self.labels[index].copy()
+        segments = self.segments[index].copy()
+        return {'img': img, 
+                'im_file': self.im_files[index], 
+                'cls': bboxes[:, 0], 
+                'instances': Instances(bboxes[:,1:], segments, bbox_format='xywh', normalized=True), 
+                'ori_shape': (h0,w0), 
+                'resized_shape': (h, w)}
+    
+    def build_transforms(self, hyp=None):
+        """Builds and appends transforms to the list."""
+        if self.augment:
+            hyp['mosaic'] = hyp['mosaic'] if self.augment and not self.rect else 0.0
+            hyp['mixup'] = hyp['mixup'] if self.augment and not self.rect else 0.0
+            transforms = v8_transforms(self, self.img_size, hyp)
```

Comment:
```suggestion
            transforms = v9_transforms(self, self.img_size, hyp)
```
Looks like there is a typo here ^

Reply:
Thanks for your reminder!